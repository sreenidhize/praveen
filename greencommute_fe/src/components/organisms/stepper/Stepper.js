import React, { useState } from "react";
import { Box, Divider, Grid, makeStyles } from "@material-ui/core";
import MStepper from "@material-ui/core/Stepper";
import Step from "@material-ui/core/Step";
import StepButton from "@material-ui/core/StepButton";
import Typography from "@material-ui/core/Typography";
import Button from "../../atom/button/Button";
import BackArrow from "../../atom/BackArrow/BackArrow";
import Autocomplete from "@material-ui/lab/Autocomplete";
import TextField from "@material-ui/core/TextField";
import Chip from "@material-ui/core/Chip";
import LocationOnOutlinedIcon from "@material-ui/icons/LocationOnOutlined";
import PropTypes from "prop-types";

const useStyles = makeStyles((theme) => ({
  textFieldStyle: {
    borderRadius: "10px",
  },
  step: {
    width: "100%",
  },
  headline: {
    width: "30rem",
    fontWeight: 500,
    margin: "1rem 0",
    fontSize: "32px",
  },
  subHeadline: {
    fontWeight: 500,
    margin: "1rem 0",
    fontSize: "20px",
  },
  viewText: {
    fontSize: "20px",
  },
  buttonMargin: {
    margin: "1rem 0.5rem",
  },
  gridContainerMargin: {
    margin: "1rem 0",
  },
  gridItemMargin: {
    margin: "0 1rem",
  },
  autoComplete: {
    borderRadius: "10px",
    width: "30rem",
  },
  chipStyle: {
    borderRadius: "6px",
  },
  inputIcon: {
    margin: "0 0.5rem",
  },
}));

const cities = [
  { city: "Hyderabad" },
  { city: "Banglore" },
  { city: "Mumbai" },
  { city: "Delhi" },
  { city: "Chennai" },
  { city: "Vizag" },
];

const skills = [
  { skill: "User Experience Designer" },
  { skill: "Production Designer" },
  { skill: "ReactJS Dev" },
  { skill: "Java Dev" },
  { skill: "Scrum Master" },
];

const getSteps = () => ["Your Location", "Job Location", "Your Skills"];

const StepperInput = (props) => {
  const classes = useStyles();
  const {
    isMulitple = false,
    accessKey,
    options,
    placeholder,
    handleChange,
    isHomeLocation = false,
    defaultValue,
  } = props;

  const getParams = (params) => {
    const { InputProps } = params;

    if (isMulitple === false) {
      return {
        ...params,
        InputProps: {
          ...InputProps,
          startAdornment: (
            <LocationOnOutlinedIcon
              color="secondary"
              className={classes.inputIcon}
            />
          ),
          classes: { root: classes.textFieldStyle },
        },
      };
    }

    return {
      ...params,
      InputProps: {
        ...InputProps,
        startAdornment: (
          <>
            <LocationOnOutlinedIcon
              color="secondary"
              className={classes.inputIcon}
            />
            {InputProps.startAdornment}
          </>
        ),
        classes: { root: classes.textFieldStyle },
      },
    };
  };

  return (
    <Autocomplete
      data-testid="autocomplete"
      id="autocomplete"
      multiple={isMulitple}
      defaultValue={defaultValue}
      className={classes.autoComplete}
      classes={{ ...classes.root }}
      variant="primary"
      options={options}
      limitTags={2}
      popupIcon={null}
      onChange={(event, value) =>
        handleChange(value, accessKey, isHomeLocation)
      }
      getOptionLabel={(option) => option[accessKey]}
      renderTags={(value, getTagProps) => {
        return value.map((option, index) => {
          return (
            <Chip
              key={index}
              classes={{
                root: classes.chipStyle,
              }}
              variant="outlined"
              color="primary"
              label={option[accessKey]}
              {...getTagProps({ index })}
            />
          );
        });
      }}
      renderInput={(params) => {
        const newParams = getParams(params);

        return (
          <TextField
            {...newParams}
            variant="outlined"
            placeholder={placeholder}
          />
        );
      }}
    />
  );
};

const renderView = (activeStep, handleChange, defaultValue) => {
  const classes = useStyles();

  if (activeStep === 0) {
    return (
      <div>
        <Typography variant="body1" className={classes.subHeadline}>
          Where do you stay?
        </Typography>
        <StepperInput
          options={cities}
          accessKey="city"
          placeholder="Your Location"
          isHomeLocation={true}
          handleChange={handleChange}
          defaultValue={defaultValue}
        />
      </div>
    );
  }

  if (activeStep === 1) {
    return (
      <>
        <Typography variant="body1" className={classes.subHeadline}>
          Where do you want to work?
        </Typography>
        <StepperInput
          isMulitple={true}
          options={cities}
          accessKey="city"
          isHomeLocation={false}
          placeholder="Job Location"
          handleChange={handleChange}
          defaultValue={defaultValue}
        />
      </>
    );
  }

  return (
    <div>
      <Typography variant="body1" className={classes.subHeadline}>
        What do you want to do?
      </Typography>
      <StepperInput
        isMulitple={true}
        options={skills}
        accessKey="skill"
        placeholder="Skills"
        handleChange={handleChange}
        defaultValue={defaultValue}
      />
    </div>
  );
};

const Stepper = ({ handleChange, handleClick, handleRoute, defaultValue }) => {
  const [activeStep, setActiveStep] = useState(0);
  const steps = getSteps();
  const classes = useStyles();

  const handleSteps = (step) => {
    if (step >= 0 && step <= 2) {
      setActiveStep(step);
      handleClick(step);
    }
  };

  return (
    <Grid container direction="column">
      <Grid item xs={12} className={classes.step}>
        <MStepper nonLinear connector={null} activeStep={activeStep}>
          {steps.map((label, idx) => (
            <Step
              key={label}
              color={idx <= activeStep ? "primary" : "secondary"}
              className={classes.step}
            >
              <StepButton completed={idx <= activeStep}>
                <Typography
                  variant="h4"
                  color={idx <= activeStep ? "primary" : "secondary"}
                >
                  {label}
                </Typography>
              </StepButton>
            </Step>
          ))}
        </MStepper>
      </Grid>
      <Grid item>
        <Divider />
      </Grid>

      <Box marginTop="7%" marginLeft="9.5%" marginBottom="4%">
        <Grid item>
          <Typography variant="h2" className={classes.headline}>
            More than 2000 people are using Green Commute
          </Typography>
        </Grid>
      </Box>
      <Box marginLeft="9.5%">
        <Grid item>
          {activeStep !== 0 && (
            <BackArrow
              className={{ ...classes.gridContainerMargin }}
              label="Back"
              onClick={() => handleSteps(activeStep - 1)}
            />
          )}
        </Grid>
        <Grid item>{renderView(activeStep, handleChange, defaultValue)}</Grid>
        <Box marginTop="2%">
          <Grid container className={classes.gridContainerMargin}>
            {activeStep !== 2 ? (
              <>
                <Grid item className={{ ...classes.gridItemMargin, margin: 0 }}>
                  <Button
                    variant="contained"
                    label="Next"
                    type="small"
                    onClick={() => handleSteps(activeStep + 1)}
                  />
                </Grid>
                <Grid item className={classes.gridItemMargin}>
                  <Button
                    variant="outlined"
                    label="Skip"
                    type="small"
                    onClick={() => handleSteps(activeStep + 1)}
                  />
                </Grid>
              </>
            ) : (
              <Grid item className={{ ...classes.gridItemMargin, margin: 0 }}>
                <Button
                  variant="contained"
                  label="Finish"
                  type="small"
                  onClick={() => handleRoute()}
                />
              </Grid>
            )}
          </Grid>
        </Box>
      </Box>
    </Grid>
  );
};

Stepper.propTypes = {
  handleChange: PropTypes.func,
  handleClick: PropTypes.func,
  handleRoute: PropTypes.func,
};
export default Stepper;
