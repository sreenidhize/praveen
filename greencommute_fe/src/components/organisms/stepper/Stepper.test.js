import React from "react";
import Stepper from "./Stepper";
import { render } from "@testing-library/react";

describe("component renders correctly", () => {
  test("should match snapshot", () => {
    const { asFragment, debug } = render(<Stepper />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("block should not be empty", () => {
    const { debug, container } = render(<Stepper />);

    // expect(container.getElementsByTagName("div").length).toBe(20);
  });
});
