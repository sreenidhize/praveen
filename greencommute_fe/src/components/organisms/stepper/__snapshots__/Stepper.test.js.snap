// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
  >
    <div
      class="MuiGrid-root makeStyles-step-2 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiPaper-root MuiStepper-root MuiStepper-horizontal MuiPaper-elevation0"
      >
        <div
          class="MuiStep-root MuiStep-horizontal makeStyles-step-2"
          color="primary"
        >
          <button
            class="MuiButtonBase-root MuiStepButton-root MuiStepButton-horizontal"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal"
            >
              <span
                class="MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active MuiStepIcon-completed"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiStepLabel-completed MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4 MuiTypography-colorPrimary"
                  >
                    Your Location
                  </h4>
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root MuiStepButton-touchRipple"
            />
          </button>
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal makeStyles-step-2"
          color="secondary"
        >
          <button
            class="MuiButtonBase-root MuiStepButton-root MuiStepButton-horizontal"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal"
            >
              <span
                class="MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    2
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4 MuiTypography-colorSecondary"
                  >
                    Job Location
                  </h4>
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root MuiStepButton-touchRipple"
            />
          </button>
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal makeStyles-step-2"
          color="secondary"
        >
          <button
            class="MuiButtonBase-root MuiStepButton-root MuiStepButton-horizontal"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal"
            >
              <span
                class="MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    3
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4 MuiTypography-colorSecondary"
                  >
                    Your Skills
                  </h4>
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root MuiStepButton-touchRipple"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <hr
        class="MuiDivider-root"
      />
    </div>
    <div
      class="MuiBox-root MuiBox-root-12"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h2
          class="MuiTypography-root makeStyles-headline-3 MuiTypography-h2"
        >
          More than 2000 people are using Green Commute
        </h2>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-13"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      />
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div>
          <p
            class="MuiTypography-root makeStyles-subHeadline-4 MuiTypography-body1"
          >
            Where do you stay?
          </p>
          <div
            aria-expanded="false"
            class="MuiAutocomplete-root makeStyles-autoComplete-9 MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
            data-testid="autocomplete"
            role="combobox"
            variant="primary"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root makeStyles-textFieldStyle-1 MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-inputIcon-11 MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zM7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 2.88-2.88 7.19-5 9.88C9.92 16.21 7 11.85 7 9z"
                  />
                  <circle
                    cx="12"
                    cy="9"
                    r="2.5"
                  />
                </svg>
                <input
                  aria-autocomplete="list"
                  aria-invalid="false"
                  autocapitalize="none"
                  autocomplete="off"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                  id="autocomplete"
                  placeholder="Your Location"
                  spellcheck="false"
                  type="text"
                  value=""
                />
                <div
                  class="MuiAutocomplete-endAdornment"
                >
                  <button
                    aria-label="Clear"
                    class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                    tabindex="-1"
                    title="Clear"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-label="Open"
                    class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                    tabindex="-1"
                    title="Open"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    />
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-14 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-15"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-18"
      >
        <div
          class="MuiGrid-root makeStyles-gridContainerMargin-7 MuiGrid-container"
        >
          <div
            class="MuiGrid-root 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-base-19 makeStyles-small-20 MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Next
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root makeStyles-gridItemMargin-8 MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-base-19 makeStyles-small-20 MuiButton-outlinedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Skip
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
